addons/mail/static/src/components/notification_group/notification_group.js

```diff
@@ -61,6 +61,9 @@ class NotificationGroup extends Component {
             return;
         }
         this.group.openDocuments();
+        if (!this.env.messaging.device.isMobile) {
+            this.env.messaging.messagingMenu.close();
+        }
```

Comment:
As discussed and as per goal of the PR.

Not in openDocuments because the action of closing the messaging menu is flux specific.
Not in models because in this case it would just be equivalent to create an onClick indirection in model and nothing else.